cross-validation/logistic-regression/script.whizzml

```diff
@@ -0,0 +1,404 @@
+                             " choosable fields.")
+              "code" 106}))))
+
+;; get-objective-name
+;;
+;; Returns the name of the field used as objective field
+;;
+;; Inputs:
+;;  dataset: (map) Dataset resource info
+;;  objective-id: (string) ID of the objective field
+;;
+;; Outputs: (string) Name of the objective field
+
+(define (get-objective-name dataset objective-id)
+  (let (fields (get dataset "fields" {}))
+    (get-in fields [objective-id "name"] false)))
+
+
+;; create-k-folds
+;;
+;; creating k-fold splits from a dataset
+;;
+;; Inputs:
+;;   dataset-id: (string) Dataset ID
+;;   k-folds: (integer) Number of folds
+;;
+;; Output: (list) List of dataset IDs
+;;
+(define (create-k-folds dataset-id k-folds)
+  (let (k-fold-fn (lambda (x) (create-dataset
+                               {"origin_dataset" dataset-id
+                                "row_offset" x
+                                "row_step" k-folds
+                                "new_fields" [{"name" "k_fold"
+                                               "field" (str x)}]}))
+        dataset-ids (map k-fold-fn (range 0 k-folds)))
+    (wait* dataset-ids)))
+
+;; pair-k-folds
+;;
+;; Builds a list of pairs of hold-out and complementary datasets for all
+;; the k-fold dataset IDs.
+;;
+;; Inputs:
+;;   dataset-ids: (list) List of the k-fold dataset IDs
+;;
+;; Output: (list) List of pairs [hold-out dataset, multidataset with the rest]
+;;
+(define (pair-k-folds dataset-ids)
+  (map (lambda(x) (list (nth dataset-ids x)
+                        (concat (take x dataset-ids)
+                                (drop (+ x 1) dataset-ids))))
+       (range 0 (count dataset-ids))))
+
+
+;; select-keys
+;;
+;; Filters the keys in a map, keeping only the ones that appear in the list.
+;;
+;; Inputs:
+;;   map: (map) Key, value maps
+;;   keys-list: (list) List of keys to be kept in the map
+;; Output: (map) filtered map with only the keys in the keys-list
+;;
+(define (select-keys map keys-list)
+  (reduce (lambda (x y) (let (value (get map y false))
+                             (cond value (assoc x y value) x)))
+          {}
+          keys-list))
+
+;; create-k-models
+;;
+;; Creates the models for a set of k-fold datasets
+;;
+;; Inputs:
+;;   type: (string) type of model (model or ensemble)
+;;   multidatasets: (list) List of lists of datset IDs once a k-fold is
+;;                         excluded
+;;   objective-name: (string) name of the objective field
+;;   model-options: (map) Options for the model or ensemble
+;;
+;; Output: (list) model IDs
+;;
+(define (create-k-models type multidatasets objective-name model-options)
+  (let (models (map (lambda (x) (create type
+                                    (merge {"datasets" x
+                                            "objective_field"
+                                            objective-name}
+                                            model-options)))
+                           multidatasets))
```

Comment:
indentation is off in several spots:

```
(map (lambda (x)
       (create type
               (merge {"datasets" x
                       "objective_field" objective_name}
                      model-options)))
     multidatasets)
```

if you're using these models directly to create evaluations without
any fetch in between, you don't need to `wait*` on them.